src/web/client/src/components/Register/Register.tsx

```diff
@@ -0,0 +1,154 @@
+import React, {
+  ChangeEvent,
+  Component,
+  ComponentState,
+  FormEvent,
+} from "react";
+import { Redirect, Link as RouterLink } from "react-router-dom";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import Container from "@material-ui/core/Container";
+
+interface Props {
+  isLoggedIn: boolean;
+  history: any;
+  onAuthChange: (authState: boolean) => void;
+}
+interface State {
+  email: string;
+  username: string;
+  password: string;
+}
+
+/**
+ * A register form for users to register an account
+ */
+export default class Register extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      email: "",
+      username: "",
+      password: "",
+    };
+  }
+
+  /**
+   * Sets the appropriate state when a keyboard event is triggered
+   * @params event - An event that records changes happening to an input field
+   */
+  onChange = (event: ChangeEvent<HTMLInputElement>): void => {
+    event.preventDefault();
+    this.setState({
+      [event.target.name]: event.target.value,
+    } as ComponentState);
+  };
+
+  /**
+   * Submits the user credentials to the auth server
+   * @params event - An event that records submission of a form element
+   *
+   * Changes the root level isLoggedIn state to True and redirects to the
+   * "/streams" route if register submission is successful.
+   * Otherwise, alert the user that register is not successful.
+   */
+  onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    try {
+      event.preventDefault();
+      const res = await fetch("/auth/register", {
+        method: "POST",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          email: this.state.email,
+          username: this.state.username,
+          password: this.state.password,
+        }),
+      });
+      const msg = await res.text();
+      // Did not Successfully register
+      if (msg === "Bad username") {
+        console.log("no good");
+        alert(msg);
+      }
+      // Successfully registered
+      else {
+        this.props.onAuthChange(true);
+        this.props.history.push("/streams");
+      }
+    } catch (error) {
+      console.log(error);
+      alert(error);
```

Comment:
console.error and what is up with alerts

Reply:
yes